internal/broker/broker.go

```diff
@@ -368,22 +368,28 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	authDone := make(chan struct{})
 	var access, data string
 	go func() {
-		access, data = b.handleIsAuthenticated(ctx, &session, authData)
+		access, data = b.handleIsAuthenticated(ctx, &session, authData).result()
```

Comment:
Following my comment on the wrapping, here would become something like: 
`result := b.handleIsAuthenticated(ctx, &session, authData)`